#! /bin/sh /usr/share/dpatch/dpatch-run
## 51_hardy.dpatch by jre <jre-phoenix@users.sourceforge.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Workaround the hardy LSB init-functions bug "Stopping a stopped daemon
## DP: returns 3 instead of 0."
@DPATCH@
diff -urNad blockcontrol~/blockcontrol.lib blockcontrol.lib
--- blockcontrol~/blockcontrol.lib	2009-01-09 16:08:16.000000000 +0100
+++ blockcontrol.lib	2009-01-09 16:38:32.000000000 +0100
@@ -978,6 +978,7 @@
 		killproc $DAEMON
 	fi
 	RETVAL=$?
+	[ "$RETVAL" -eq 3 ] && RETVAL=0
 	log_end_msg $RETVAL
 }
 
